src/Datastore/DatastoreClient.php

```diff
@@ -0,0 +1,796 @@
+     * @return string The updated entity version number
+     */
+    public function delete(Key $key, array $options = [])
+    {
+        $res = $this->deleteBatch([$key], $options);
+        return $this->parseSingleMutationResult($res);
+    }
+
+    /**
+     * Delete multiple entities
+     *
+     * Example:
+     * ```
+     * $keys = [
+     *     $datastore->key('Person', 'Bob'),
+     *     $datastore->key('Person', 'John')
+     * ];
+     *
+     * $datastore->deleteBatch($keys);
+     * ```
+     *
+     * @param Key[] $keys The identifiers to delete
+     * @param array $options Configuration options
+     * @return array [Response Body](https://cloud.google.com/datastore/reference/rest/v1/projects/commit#response-body)
+     */
+    public function deleteBatch(array $keys, array $options = [])
+    {
+        $this->operation->mutate('delete', $keys, Key::class);
+        return $this->operation->commit($options);
+    }
+
+    /**
+     * Retrieve an entity from the datastore
+     *
+     * Example:
+     * ```
+     * $key = $datastore->key('person', 'Bob');
+     *
+     * $entity = $datastore->lookup($key);
+     * if (!is_null($entity)) {
+     *     echo $entity['firstName']; // 'Bob'
+     * }
+     * ```
+     *
+     * @codingStandardsIgnoreStart
+     * @param Key $key $key The identifier to use to locate a desired entity
+     * @param array $options {
+     *     Configuration Options
+     *
+     *     @type string $readConsistency See
+     *           [ReadConsistency](https://cloud.google.com/datastore/reference/rest/v1beta3/ReadOptions#ReadConsistency).
+     *     @type string $className The name of the class to return results as.
+     *           Must be a subclass of {@see Google\Cloud\Datastore\Entity}.
+     *           If not set, {@see Google\Cloud\Datastore\Entity} will be used.
+     * }
+     * @return Entity|null
+     * @codingStandardsIgnoreEnd
+     */
+    public function lookup(Key $key, array $options = [])
+    {
+        $res = $this->lookupBatch([$key], $options);
+
+        return (isset($res['found'][0]))
+            ? $res['found'][0]
+            : null;
+    }
+
+    /**
+     * Get multiple entities
+     *
+     * Example:
+     * ```
+     * $keys = [
+     *     $datastore->key('Person', 'Bob'),
+     *     $datastore->key('Person', 'John')
+     * ];
+     *
+     * $entities = $datastore->lookup($keys);
+     *
+     * foreach ($entities['found'] as $entity) {
+     *     echo $entity['firstName'];
+     * }
+     * ```
+     *
+     * @codingStandardsIgnoreStart
+     * @param Key[] $key The identifiers to look up.
+     * @param array $options {
+     *     Configuration Options
+     *
+     *     @type string $readConsistency See
+     *           [ReadConsistency](https://cloud.google.com/datastore/reference/rest/v1beta3/ReadOptions#ReadConsistency).
+     *     @type string $className The name of the class to return results as.
```

Comment:
If we have two keys with different kinds:

```php
$keys = [
    $datastore->key('Person', 'John'),
    $datastore->key('Location', 'Home')
];
```

would we need to allow setting the `className` for each?